spec/integration/rails_helper_spec.rb

```diff
@@ -71,23 +62,29 @@ def tag_options(options)
     sort("other")
   end
 
+  it "should set :href on anchor" do
+    should_receive(:url_for).and_return('/example')
+    should_receive(:content_tag).with(:a, 'Field', href: '/example')
+    sort("field")
+  end
+
   it "should add no styling by default" do
     should_receive(:url_for)
-    should_receive(:a_link).with('Field', anything, hash_excluding(:class))
+    should_receive(:content_tag).with(:a, 'Field', hash_excluding(:class))
     sort("field")
   end
 
   it "should add ascending style for current ascending sort order " do
     should_receive(:url_for)
-    should_receive(:a_link).with('&#9650;&nbsp;Field', anything, hash_including(:class => 'ascending'))
+    should_receive(:content_tag).with(:a, '&#9650;&nbsp;Field', hash_including(:class => 'ascending'))
 
     params[:order] = "field ASC"
     sort("field")
   end
 
   it "should add descending style for current descending sort order " do
     should_receive(:url_for)
-    should_receive(:a_link).with('&#9660;&nbsp;Field', anything, hash_including(:class => 'descending'))
+    should_receive(:content_tag).with(:a, '&#9660;&nbsp;Field', hash_including(:class => 'descending'))
```

Comment:
Yeah, I think I see your point, though I would be hesitant to match the generated HTML too closely in case the implementation of the tag/content_tag builder changes - even reordering of attributes in the tag could break regex matching. This shouldn't really be a test of `content_tag`.

I can remove expectations and match the generated tag to regular expressions of the HTML, or add a dep on a HTML parser (similar to the Rails test helper `assert_tag`).